### PR TITLE
Adding a map for point -> node for faster lookup

### PR DESCRIPTION
**Pull request by RiskRunner0**
Still todo: implement a list in the game for everything that needs to be painted.  Main and View are so tightly coupled, the logic is hard to follow, we need to figure out how to make that better.

### CODE REVIEW

**Comment by hippi345** [comment]
I am not sure on the existing conflicts, the one in Game I feel makes sense to just integrate and merge but in Main I am not sure if you were wanting to change the method name or remove the original method for treasure detection and change to status determination.
@QueenCitySmitty

**Comment by RiskRunner0** [comment]
I'll deal with the conflicts tonight, I had these 2 separate work streams and was going to deal with the merge issues whenever one of the two went in.

File: src/sample/Main.java [hunk 1]
```diff
@@ -29,9 +29,6 @@ public void start(Stage primaryStage)
 
     static void setupView(View gameView)
     {
-        // creation of the grid pane
-        gameView.setupTheGridPane();
-
         // setting up the buttons which go into the UI
         gameView.setupButtons(
```
Comment (hippi345) [inline src/sample/Main.java:33]: We do need to make note to work getting this pattern completely out of the project. It is getting there.

Comment (hippi345) [reply src/sample/Main.java:33]: But also the two remaining ones are not even related. one is make move and the other is return to startup gui so I would say if we do not have a task for this then we need to.

File: src/sample/View.java [hunk 2]
```diff
@@ -144,10 +149,16 @@ private void startGameGUI(int parseInt) {
...
+        // creation of the grid pane
+        gameView.setupTheGridPane();
+
         // preparation of the game components for movement and treasure hunting
         Main.prepareGame();
-        // establishes the actual View for the game which is managed in the View class
+
         Main.setupView(gameView);
+
         // size based on arg algorithm
```
Comment (hippi345) [inline src/sample/View.java:162]: this needs to be more sophisticated and less guesswork. it should ideally be based on the size of the machine's screen.

File: src/sample/View.java [hunk 3]
```diff
@@ -156,32 +167,38 @@ private void startGameGUI(int parseInt) {
...
-                else if(treasure.getX() == currentColumnIndex && treasure.getY() == currentRowIndex)
-                {
-                    textToSet = treasure.getX() + " " + treasure.getY() + " " + " treasure";
-                }
-                else
-                {
-                    textToSet = currentColumnIndex + " " + currentRowIndex + " empty";
-                }
-                ((Text) node).setText(textToSet);
+                ((Text) node).setText(currentColumnIndex + " " + currentRowIndex + " empty");
```
Comment (hippi345) [inline src/sample/View.java:178]: you love to see that finally die lol gg

**Review by hippi345 — APPROVED** [review]
